src/hooks/useTheme.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect, useCallback } from "react";
+
+const THEME_KEY = "darkMode";
+
+export function useTheme() {
+    const getInitialMode = () => {
+        const savedTheme = localStorage.getItem(THEME_KEY);
+        if (savedTheme !== null) {
+            return savedTheme === "true";
+        }
+        return window.matchMedia?.("(prefers-color-scheme: dark)")?.matches ?? false;
```

Comment:
If used in SSR, localStorage and window won't be defined. Always guard against it.